src/ClientContextInterface.php

```diff
@@ -136,23 +136,23 @@
  * @method $this ftsynupdate(string $index, string $synonymGroupId, ?SynUpdateArguments $arguments = null, string ...$terms)
  * @method $this fttagvals(string $index, string $fieldName)
  * @method $this get($key)
- * @method $this getbit($key, $offset)
- * @method $this getex(string $key, $modifier = '', $value = false)
- * @method $this getrange($key, $start, $end)
+ * @method $this getbit(string $key, int $offset)
+ * @method $this getex(string $key, string $modifier = '', string|int $value = false)
+ * @method $this getrange(string $key, int $start, int $end)
  * @method $this getdel(string $key)
- * @method $this getset($key, $value)
+ * @method $this getset(string $key, string|int|float $value)
  * @method $this incr($key)
  * @method $this incrby($key, $increment)
  * @method $this incrbyfloat($key, $increment)
  * @method $this mget(array $keys)
  * @method $this mset(array $dictionary)
  * @method $this msetnx(array $dictionary)
- * @method $this psetex($key, $milliseconds, $value)
- * @method $this set($key, $value, $expireResolution = null, $expireTTL = null, $flag = null)
- * @method $this setbit($key, $offset, $value)
- * @method $this setex($key, $seconds, $value)
- * @method $this setnx($key, $value)
- * @method $this setrange($key, $offset, $value)
+ * @method $this psetex(string $key, int $milliseconds, string|int|float $value)
+ * @method $this set(string $key, string|int|float $value, ?string $expireResolution = null, ?int $expireTTL = null, ?string $flag = null)
+ * @method $this setbit(string $key, int $offset, string|int|float $value)
+ * @method $this setex(string $key, int $seconds, string|int|float $value)
```

Comment:
This might be a bit strict. Keys could be numeric as well and get casted to a string no? Same for values, could they not be objects that implement `__toString()`?

Reply:
Yeah, you're right, I double checked and it's possible to use integers as keys as they anyway will be encoded as bulk strings. And yeah, it's possible to pass objects that implements `__toString()`, it's even possible to pass boolean as value however string conversion works non-intuitive (true -> '1', false -> ''), the only exception is arrays. I think it makes sense to mark all of it as `mixed`?

Reply:
I think explicit `(string)` cast should be done on user-side. If you go this way all way down, then values can also be `null` `bool` or stringify-able objects, so you end up with `mixed` everywhere which adds no value.

And BTW `mixed` will allow `array` while it's not actually valid.

Reply:
We might need to revise #1521 as well @vladvildanov.

Reply:
`mixed` is what Relay used as well, I'm not sure it's worth adding for key-names and values TBH because it's implied already.

Reply:
@tillkruss In this way it will be inconsistent to all other commands, if you walk through the interface case are usually typed as strings

Reply:
My vote is to not touch the stubs, it's been working and restricting them will unleash dozens of tickets I don't want to tackle.